File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        n = len(nums) #len of the arr
        k %= n

        self.reverse(nums, n - k, n - 1)
        self.reverse(nums, 0, n - k - 1)
        self.reverse(nums, 0, n - 1)

    def reverse(self, nums, start, end):
        while start <= end:
            nums[start], nums[end] = nums[end], nums[start]
            start += 1
            end -= 1

        
        

File: 0189-rotate-array/0189-rotate-array.java
class Solution {

    public void reversearrar(int[] nums, int start, int end) {
        while (start < end) {
            int temp = nums[start];
            nums[start] = nums[end];
            nums[end] = temp;
            start++;
            end--;

        }
    }

    public void rotate(int[] nums, int k) {
        int n = nums.length;
        k = k % n;

        reversearrar(nums, n-k, n-1);
        reversearrar(nums, 0, n - k - 1);
        reversearrar(nums, 0, n - 1);
    }
}